src/js_native_api_v8.cc

```diff
@@ -2526,15 +2526,16 @@ napi_status NAPI_CDECL napi_get_value_string_utf16(napi_env env,
     // V8 assumes UTF-16 length is the same as the number of characters.
     *result = val.As<v8::String>()->Length();
   } else if (bufsize != 0) {
-    int copied = val.As<v8::String>()->Write(env->isolate,
-                                             reinterpret_cast<uint16_t*>(buf),
-                                             0,
-                                             bufsize - 1,
-                                             v8::String::NO_NULL_TERMINATION);
+    size_t n_chars = std::min(
```

Comment:
nit: the `n_chars` is a bit cryptic and unusual in this context. Could you use something like `length`, `str_length`, `str_len`, or `str_size`. The `Write` function parameter name is `length`.